### DISCUSSION TITLE
Currently playing video in VLC playlist?

### DISCUSSION THREAD

**Discussion by therentabrain** [WarmUpTill/SceneSwitcher#1202] [Q&A]
Is there a way to get the currently playing file in a VLC video playlist source?

**Comment by therentabrain** [comment]
I have found a wackadoodle Windows workaround in case there isn't a more direct way.
Using [Handle](https://learn.microsoft.com/en-us/sysinternals/downloads/handle), you can specify the process and a match string, like:

`handle.exe -nobanner -vt -p obs64 "showvideos\\"`

This would show any OBS in-use files in the showvideos folder.

Then it would just be a question of parsing that string to get the filename.  In my case, I don't mind the messy output because it's just for my own information, to find out what randomly chosen media file is playing.  

Still, it would be nice to be able to get this kind of information and display it in a nicer manner, all in OBS nice and neat.  Is there a way?

Thanks!

**Comment by WarmUpTill** [comment]
This is not implemented at the moment, but I will try to look into it during the weekend.
I think it _should_ be possible at least with the VLC source type as far as I can tell as it offers a `proc_handler` to query metadata.

**Reply by WarmUpTill** [reply]
A build with the option to assign the title of playlist item currently being played by a VLC source to a variable will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/10919347213?pr=1205

![image](https://github.com/user-attachments/assets/d0ca5c12-35a5-483c-90fb-4712e4223410)

Let me know if that resolves your problem :)

**Reply by therentabrain** [reply]
So amazing.  People have been looking for this for years.  I'll try it out and let you know ASAP

**Comment by therentabrain** [comment]
Working on this now, but just in case you have a sec to advise:  If I wanted to create a log file (in Windows) of the videos and what time they played, how would I do it?  

What comes to mind is to set the variable, then run a cmd that echoes the variable and a CR+LF to a text file.

But, I take it there isn't currently an event/condition for "video file change".  Of course I could have SS check every 5 seconds and send the value to a program (like in vbscript) that only logs when the argument it receives has changed, but that seems pretty inefficient.  Some files are 20-30 seconds long, and some are around twenty minutes.  And I'm only playing videos for about 50% of the stream time.

It's tempting to ditch VLC Playlists entirely and play files one at a time using SS to monitor for when they end and trigger the next one.    That's a bunch of wheel to reinvent though.

Is there a way to say "when one of these VLC Sources is active, log every filename when it changes"?

I wish the VLC Source could have its own "Save log to this location" property.   Even a "Log VLC activity to OBS log" would be better than nothing.

Another thought:  If I had a "macro filter" which I could add to a bunch of Sources, where SS would run specific macros on that Source.  Like a mini SS that lives in the properties of that filter. :)  In case that idea means something :)

I'll think on it!

**Comment by WarmUpTill** [comment]
> Is there a way to say "when one of these VLC Sources is active, log every filename when it changes"?

I think you can achieve this with two macros like this:

1. A macro which checks that the source is active / playing something and assigns the current title to a variable:
![](https://github.com/user-attachments/assets/194af114-dbf1-4085-ad6b-d8e469615af6)
2. A macro which check if the title changes and writes it to the OBS log.
![](https://github.com/user-attachments/assets/5a71cedd-0cfb-4c2b-96cc-4abf627a2159)

You can also write to a custom file instead using the `File` action type if that should be preferred.

You can [import](https://github.com/WarmUpTill/SceneSwitcher/wiki/Exporting-and-importing-individual-macros) this example using this code:
```
AAAO+3ic7VdLbxs3EP4rBNFDm3rdXVkPS7fUcVABtqNajnoojILaHa0Ir8jFkqvINfTfO0PuSxYSuA3qXHIQsBzOi998M6Se+EbEhTZ88ucTV2IDfMKvdMryTDxm0lgmLWwMP+G5KA1urkRmgFaFyDLIGoFWF2uhUtSwRYlr8yDzyx3EH9TcisI2esbq/G1spVZmurrR9p1Wrde00GVee3CLd8IKPnnisc4ykRtIWkfybzTs7fcnTd4Xa4gfmNFlEQMzVlj4N3nXfr8i8WqV6PhhDtZKlRpKvoAUkYSiUVgLc1uqX0trtaqPi7IZpXokxTRsaa7EspP1WqbrDH92urrQKpEurTu0aDSK2v8d7PAQHOPVYBzIXUzcKdXx3kcvw934IIjPqFJq4zNRAKO0Tw8s3lNCXzJxGZMN4TbFCnORbI0JHDEDEgYRxFEUw1mwDMdJ0B+NouA8jAbBYChADKJwPDwP0cEWCiMJvGh/4nntK9YEnOeZtFiJmTZu7VifE/t7YUhMct/ReLy/rxxcYm7eyedto+GosQ3Jsi74b9o+wKPpsA/h9EK0vm9Q74qsTtMMZs/FLZ4uroF0A8p2WXbcIujiokTebg7JE3dl/PqRZe4TIZME/gYSKRDMTKcyxgNhXcpCUOhrnciVJB4/cSs38BcmZWwhpLJOzwBl6ZLZiozI2HyEp7hvH3P63NO5pTfpVAzFvnWde696Urf24uqCLWQCms29DmnHoGAOGcS+HMdWvPF5V+/ENCOalR8Sk4hKhieRlaez9sz/8TAHCyyfiF+ldltRSLHMqGVB0UdSD5J658YjcydtBmzRqtf7vUphqvCwMmFOni1xnjZAE2zFwkPBcaXKzaIDTMNUaoyT+rTTZMcnAQJtyuUckVZpNV7DrsjN9NA1EOxmgtoN3fDTN7ySXQsbr50v1FpJldySlCD05+0OeouJt3M+rwpAFSNDjEg44ERiVjOSuBh478VwuFcJ2Sdp13UeOFFWMv1SXA/7QdiNsOvLXY5U88zgP7KI/cx67A07Yz+xX1ifdwo/K/Qmb+8fqfLS1jL+Fl2kijlCUo51+Sbew9QpU9ZrnSWdm0cebfAgwPJAwSweNwgcc7ay0Iqqtjhkzcf55e3s9sP76dUlf2n/4fMhRzftdkRc9QV0nLVExefD2/meou1UJb6+Ve9F3c5LZFEHd81MHLoClWKdPmORiyRBJbzzMaP+ObJVmHo4UJ9CM+6Nn7sOsBoGJ+s8OP4o8HVE8OOsfN0n0veXxv/00kj6ZyDCPgS989Uo6ItwFIh4PAwgHkbJaByNVuMjsn5/aXz1bfUNHhvf5EYcvPj+OL63XvEd4efZsycECq/x6hI0wfiN/uT+I2LkCfvh6RC9PX/pLL1vcT74C3pUDSO2lTPPEliJMrMN5i02v5dQQhWwKTiPTnuj014wDNLRchwO4sGA7/8BNlsqxw==
```

Hope that helps!
